src/pcap/agent.go

```diff
@@ -115,7 +115,7 @@ func (a *Agent) Capture(stream Agent_CaptureServer) (err error) {
 		return errorf(codes.InvalidArgument, "%w", err)
 	}
 
-	opts := req.Payload.(*AgentRequest_Start).Start.Capture
+	opts := req.Payload.(*AgentRequest_Start).Start.Capture //nolint:errcheck //this only returns one value
```

Comment:
we could have a check here for the cast.
```suggestion
	var opts *CaptureOptions
	if start, ok := req.Payload.(*AgentRequest_Start); ok {
		opts = start.Start.Capture
	} else {
		return errorf(codes.InvalidArgument, "invalid message: expected Payload of type StartAgentCapture")
	}
```
(not tested)

Reply:
Ok. Maybe we can eventually return the cast result, but not in this small linter fix